src/main/scala/scala/collection/decorators/IterableDecorator.scala

```diff
@@ -33,4 +33,46 @@ class IterableDecorator[C, I <: IsIterable[C]](coll: C)(implicit val it: I) {
   def lazyFoldRight[B](z: B)(op: it.A => Either[B, B => B]): B =
     it(coll).iterator.lazyFoldRight(z)(op)
 
+
+  /**
+    * Constructs an iterator where consecutive elements are accumulated as
+    * long as the output of f for each element doesn't change.
+    * <pre>
+    * Vector(1,2,2,3,3,3,2,2)
+    * .splitBy(identity)
+    * </pre>
+    * produces
+    * <pre>
+    * Iterator(Vector(1),
+    * Vector(2,2),
+    * Vector(3,3,3),
+    * Vector(2,2))
+    * </pre>
+    *
+    * @param f the function to compute a key for an element
+    * @tparam K the type of the computed key
+    * @return an iterator of sequences of the consecutive elements with the
+    *         same key in the original iterator
+    */
+  def splitBy[K, That, CC](f: it.A => K)(implicit bf: BuildFrom[C, it.A, That], bff: BuildFrom[C, That, CC]): CC = {
+    val builder = bff.newBuilder(coll)
+
+    val iterator = it(coll).iterator
+    var init = bf.newBuilder(coll)
+    if (iterator.hasNext) {
+      var ref = iterator.next();
+      init += ref
+      while (iterator.hasNext) {
+        val el = iterator.next();
+        if (f(el) != f(ref)) {
```

Comment:
Should we put `f(ref)` in a val to compute it only once?

Reply:
I reverted to delegating to iterator's implementation and the tests still pass. I guess I got confused with my attempts at having splitBy defined for both IterableDecorator and SeqDecorator ...

Reply:
this becomes irrelevant with the revert to delegating to iterator.